icu4c/source/test/cintltst/ucptrietest.c

```diff
@@ -705,6 +707,59 @@ testBuilder(const char *testName, const UMutableCPTrie *mutableTrie,
     testTrieGetRanges(testName, NULL, mutableTrie, UCPMAP_RANGE_NORMAL, 0, checkRanges, countCheckRanges);
 }
 
+static void
+trieTestGolden(const char *testName,
+             const UCPTrie* trie,
+             const CheckRange checkRanges[],
+             int32_t countCheckRanges) {
+    log_verbose("golden testing Trie '%s'\n", testName);
+
+    UErrorCode status = U_ZERO_ERROR;
+    const char *testdatapath = loadSourceTestData(&status);
+    char goldendatapath[512];
+    // note: snprintf always writes a NUL terminator.
+    snprintf(goldendatapath, 512, "%sucptrie%s%s.toml",
+        testdatapath, U_FILE_SEP_STRING, testName);
+
+    // Write the data into a tmpfile (memstream is not portable)
+    FILE* stream = tmpfile();
+    fputs("[ucptrie.struct]\n", stream);
```

Comment:
"code_point_trie" or "codepointtrie" instead of "ucptrie" ?

Reply:
Done